bin/perseo

```diff
@@ -140,7 +140,9 @@ function loadConfiguration() {
     }
     else if (process.env.PERSEO_SMTP_SECURE === 'false') {
         config.smtp.secure = false;
-        delete config.smtp.auth;
+        config.smtp.auth = config.smtp.auth || {};
```

Comment:
This hack is doing in 'false' case the same that does in 'true' case. Maybe the important is just initialize smtp.auth to {} instead of delete smtp.auth.

Reply:
@ccsr in order to have more insight on this... could you share also the Perseo FE configuration you are using, please?

Reply:
Below is yml file used to deploy in Kubernetes 
apiVersion: apps/v1
kind: Deployment
metadata:
  creationTimestamp: null
  labels:
    io.kompose.service: perseo-fe
  name: perseo-fe
spec:
  selector:
    matchLabels:
      io.kompose.service: perseo-fe
  replicas: 1
  strategy: {}
  template:
    metadata:
      creationTimestamp: null
      labels:
        io.kompose.service: perseo-fe
    spec:
      imagePullSecrets:
      - name: gitlab-registry
      containers:
      - env:
        - name: PERSEO_MONGO_ENDPOINT
          value: mongo
        - name: PERSEO_CORE_URL
          value: http://perseo-core.NAMESPACE.svc.cluster.local:8081
        - name: PERSEO_LOG_LEVEL
          value: debug
        - name: PERSEO_ORION_URL
          value: https://orion.NAMESPACE.svc.cluster.local:1026
        - name: PERSEO_SMTP_HOST
          valueFrom:
            secretKeyRef:
              name: email-secret
              key: host
        - name: PERSEO_SMTP_PORT
          value: "587"
        - name: PERSEO_SMTP_SECURE
          value: "false"
        - name: PERSEO_SMTP_AUTH_USER
          valueFrom:
            secretKeyRef:
              name: email-secret
              key: user
        - name: PERSEO_SMTP_AUTH_PASS
          valueFrom:
            secretKeyRef:
              name: email-secret
              key: password
        image: gitlab.opendata-_somecity.com_:5050/fiware/perseo:BRANCHNAME
        imagePullPolicy: Always
        name: perseo-fe
        volumeMounts:
        - name: perseo-config
          mountPath: /opt/perseo-fe/config.js
          subPath: config.js
        resources: {}
      volumes:
      - name: perseo-config
        secret:
          secretName: perseo-config
      restartPolicy: Always
status: {}

---

apiVersion: v1
kind: Service
metadata:
  creationTimestamp: null
  labels:
    io.kompose.service: perseo-fe
  name: perseo-fe
spec:
  ports:
  - name: "9090"
    port: 9090
    targetPort: 9090
  selector:
    io.kompose.service: perseo-fe
status:
  loadBalancer: {}

Reply:
also config.js content below

_____
/*
 * Copyright 2015 Telefonica Investigación y Desarrollo, S.A.U
 *
 * This file is part of perseo-fe
 *
 * perseo-fe is free software: you can redistribute it and/or
 * modify it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the License,
 * or (at your option) any later version.
 *
 * perseo-fe is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public
 * License along with perseo-fe.
 * If not, see http://www.gnu.org/licenses/.
 *
 * For those usages not covered by the GNU Affero General Public License
 * please contact with::[contacto@tid.es]
 */

'use strict';

var config = {};

/**
 * Default log level. Can be one of: 'DEBUG', 'INFO', 'WARN', 'ERROR', 'FATAL'
 * @type {string}
 */
config.logLevel = 'INFO';

/**
 * Configures the exposed API.
 */
config.endpoint = {
    host: 'localhost',
    port: 9090,
    rulesPath: '/rules',
    actionsPath: '/actions/do',
    noticesPath: '/notices',
    vrPath: '/m2m/vrules',
    checkPath: '/check',
    versionPath: '/version',
    logPath: '/admin/log',
    metricsPath: '/admin/metrics'
};

/**
 * Is Master this one?
 */
config.isMaster = true;

/**
 * Delay (in milliseconds) for slave to execute an action
 */
config.slaveDelay = 500;

/**
 * DB Configuration.
 */
config.mongo = {
    // The URI to use for the database connection. It supports replica set URIs.
    // mongodb://[username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]][/[database][?options]]
    // I.e.: 'mongodb://user:pass@host1:27017,host2:27018,host3:27019/cep?replicaSet=myrep'
    url: 'mongodb://localhost:27017/cep'
};

/**
 * OrionDB Configuration.
 */
config.orionDb = {
    url: 'mongodb://localhost:27017/orion',
    collection: 'entities',
    prefix: 'orion',
    batchSize: 500
};

/**
 * EPL core options
 *
 * interval is the time in milliseconds between refreshing rules
 * at core. Each <<interval>> ms, the rules are sent to core.
 */
config.perseoCore = {
    rulesURL: 'http://localhost:8080/perseo-core/rules',
    noticesURL: 'http://localhost:8080/perseo-core/events',
    interval: 60e3 * 5
};
/**
 * NEXT EPL core options (with HA)
 */
config.nextCore = {
    //
    // Note this parameter is empty, so Perseo will not use HA by default
    //
    /*
    rulesURL : 'http://next-core:8080/perseo-core/rules',
    noticesURL : 'http://next-core:8080/perseo-core/events'
    */
};

/**
 * SMTP endpoint options
 */
config.smtp = {
    port: 25,
    host: 'smtpserver',
    secure: false,
    /*
     auth: {
     user: 'abc',
     pass: 'xyz'
     }
     ,
    */
    tls: {
        // keep rejectUnauthorized to false when secure is also false
        rejectUnauthorized: false
    }
};

/**
 * SMS endpoint options
 */
config.sms = {
    URL: 'http://sms-endpoint/smsoutbound',
    API_KEY: '',
    API_SECRET: '',
    from: 'tel:22012;phone-context=+34'
};

/**
 * SMPP endpoint options
 */
config.smpp = {
    host: '',
    port: '',
    systemid: '',
    password: '',
    from: '346666666',
    enabled: false
};

/**
 * Orion (Context Broker) endpoint options
 */
config.orion = {
    URL: 'http://orion-endpoint:1026'
};

/**
 * Pep (proxy for Context Broker) endpoint options
 */
config.pep = {
    URL: 'http://pep-endpoint:1026'
};

/**
 * Authorization endpoint
 */
config.authentication = {
    host: 'keystone',
    port: '5001',
    user: 'user',
    password: 'password'
};

/**
 * Collections
 * @type {String}
 */
config.collections = {
    rules: 'rules',
    executions: 'executions'
};

/**
 * Executions TTL
 *
 * Number of seconds to expire a document in 'executions'
 * @type {Number}
 *
 */
config.executionsTTL = 1 * 24 * 60 * 60;

/**
 * Constants for missing header fields for service (Fiware-servicepath) DEFAULT_SUBSERVICE
 * and tenant (Fiware-service) DEFAULT_TENANT
 *
 * @type {String}
 */
config.DEFAULT_SUBSERVICE = '/';
config.DEFAULT_SERVICE = 'unknownt';

/**
 * CheckDB configuration
 *
 * delay Number of milliseconds to check DB connection
 * @type {Number}
 *
 * reportInterval Number of milliseconds to report a problem with DB connection in logs
 * @type {Number}
 */
config.checkDB = {
    delay: 5e3,
    reconnectTries: 1e3,
    reconnectInterval: 5e3,
    bufferMaxEntries: 5
};

/**
 * Rest base for changing access to twitter
 *
 * It should not be changed in normal conditions
 *
 * restBase:  URL
 * @type {String}
 *
 */
config.restBase = null;

/**
 * Cast attribute values in updateAction using NGSIv2 types
 */
config.castTypes = false;

module.exports = config;
_____

Reply:
Could you edit your above comments to format it a little bit, please? (code style use to be the best for configuration fragments, have a look to https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code-and-syntax-highlighting)

It is a bit hard in its current form to read...

Reply:
@fgalan I sent you email

Reply:
I see:

```
        - name: PERSEO_SMTP_SECURE
          value: "false"
```

Have you tried (in the original code, i.e. without the hack) to use `true` for that setting?

Reply:
yes, if its true we get a different error. I believe it was related to the ssl, and lead us to the old components in the package

emails did not support the old component anymore, basically

Reply:
And we are connecting to port 587, when the value should be false

Reply:
http://nodemailer.com/smtp/ - official documentation refers to using false for port 587 and starttls here